SchoolNetAssociation.UntiTest/Api/Controllers/SchoolDistrictControllerTests.cs

```diff
@@ -0,0 +1,126 @@
+			.RuleFor(sd => sd.Name, f => f.Company.CompanyName())
+			.RuleFor(sd => sd.Description, f => f.Lorem.Sentence())
+			.RuleFor(sd => sd.City, f => f.Address.City())
+			.RuleFor(sd => sd.Superintendent, f => f.Name.FullName())
+			.RuleFor(sd => sd.IsPublic, f => f.Random.Bool())
+			.RuleFor(sd => sd.NumberOfSchools, f => f.Random.Int(1, 10000))
+			.Generate(count);
+
+			return _mapper.Map<List<SchoolDistrictDto>>(fakeSchoolDistricts);
+		}
+
+		[Fact]
+		public async Task GetAllSchoolDistrictsAsync_ShouldReturnOkWithSchoolDistricts()
+		{
+			// Arrange
+			var fakeSchoolDistricts = GetFakeSchoolDistrictDtos(5);
+			_mockSchoolDistrictService.Setup(x => x.GetAllSchoolDistrictsAsync()).ReturnsAsync(fakeSchoolDistricts);
+
+			// Act
+			var result = await _controller.GetAllSchoolDistrictsAsync();
+
+			// Assert
+			_mockSchoolDistrictService.Verify(x => x.GetAllSchoolDistrictsAsync(), Times.Once);
+			result.Should().BeOfType<OkObjectResult>().Which.Value.Should().BeEquivalentTo(fakeSchoolDistricts);
+
+		}
+		[Fact]
+
+		public async Task GetSchoolDistrictByIdAsync_WithValidId_ReturnsOkResultWithSchoolDistrict()
+		{
+			// Arrange
+			var fakeSchoolDistrict = GetFakeSchoolDistrictDtos(1)[0];
+			var schoolDistrictId = ObjectId.GenerateNewId().ToString();  //_faker.Random.Guid() ->not of type ObjectId 
+			fakeSchoolDistrict.Id=schoolDistrictId;
+			_mockSchoolDistrictService.Setup(x => x.GetSchoolDistrictByIdAsync(schoolDistrictId)).ReturnsAsync(fakeSchoolDistrict);
+
+			// Act
+			var result = await _controller.GetSchoolDistrictByIdAsync(schoolDistrictId);
+
+			// Assert
+			_mockSchoolDistrictService.Verify(x => x.GetSchoolDistrictByIdAsync(schoolDistrictId), Times.Once);
+			result.Should().BeOfType<OkObjectResult>().Which.Value.Should().BeEquivalentTo(fakeSchoolDistrict);
+		}
+
+		[Fact]
+		public async Task GetSchoolDistrictByIdAsync_WithInvalidId_ReturnsBadRequest()
+		{
+			// Arrange
+			var invalidId = _faker.Name.FullName();
+
+			// Act
+			var result = await _controller.GetSchoolDistrictByIdAsync(invalidId);
+
+			// Assert
+			result.Should().BeOfType<BadRequestObjectResult>().Which.Value.Should().Be(ResponseMessages.InvalidSchoolDistrictId);
+		}
+
+		[Fact]
+		public async Task CreateSchoolDistrictAsync_WithValidData_ShouldReturnOkWithCreatedData()
+		{
+			// Arrange
+			var newSchoolDistrictDto = GetFakeSchoolDistrictDtos(1)[0];
+			_mockSchoolDistrictService.Setup(x => x.CreateSchoolDistrictAsync(newSchoolDistrictDto)).ReturnsAsync(newSchoolDistrictDto);
+
+			// Act
+			var result = await _controller.CreateSchoolDistrictAsync(newSchoolDistrictDto);
+
+			// Assert
+			_mockSchoolDistrictService.Verify(x => x.CreateSchoolDistrictAsync(newSchoolDistrictDto), Times.Once);
+			result.Should().BeOfType<OkObjectResult>().Which.Value.Should().BeEquivalentTo(newSchoolDistrictDto);
+		}
+
+		[Fact]
+		public async Task CreateSchoolDistrictAsync_WithInvalidData_ShouldReturnBadRequestWithErrors()
```

Comment:
Please tests per item as well for validation or if it’s covered under tests for validator that’s fine too.